### DISCUSSION TITLE
SoloQ / 3s PTR feedback

### DISCUSSION THREAD

**Discussion by Amrx2** [chromiecraft/chromiecraft#7632] [PTR Feedback]
As per the conversation shared in the PvP discord channel, please use this thread to give feedback on the current state of matchmaking for these modules.

Please share the comps used, how the teams were divided etc along with the talents used. There is only so much we can do on a single computer with 6 windows open so the more people who can test this, the more we can polish it and move the findings into live. Thank you.

**Comment by Ottawia** [comment]
The situation we have on the PTR: the system is waiting for 6 players: 2 melee, 2 ranged, and 2 healers. The team is still easy to predict: the first melee + the first caster + the first healer versus the second ones. Because of this, stacking the same classes still occurs (we had a boomkin in one team and shadow and disc priests in the other).

Also, one player queued without distributing talents. He was counted as melee (shaman) and had enhancement gear, which could be the reason he was queued as melee. However, fixing this issue by implementing a rule such as 'no talent points = can't queue' would be a good solution. I later tried to queue as a priest without talents and got classified as a caster, but when I did it again, I was classified as melee.

**Comment by Mikel0ke** [comment]
System is forcing 1 of each role to make the queue pop.
Every team has 1 melee 1 caster 1 healer.

**Comment by Mikel0ke** [comment]
first melee gets paired with first caster and first healer to queue, and they face second melee second caster and second healer to queue.
order in queue does not matter, if 2 casters queue first 1 will go to team1 other will go to team2. Same with melees and healers.
![image](https://github.com/user-attachments/assets/c8b9fdc1-2fb2-4e45-84ba-c9b85dd699c7)

**Comment by Ottawia** [comment]
![WoWScrnShot_121724_215140](https://github.com/user-attachments/assets/f762102e-c118-49de-ba74-dc5e2d87a379)
Low levels are displaying in arena queue but arena does not pop with them (thats good ofc) however there is no gear/gem/enchant req (we were queueing with starting zone gear mostly)

**Comment by Helias** [comment]
Thanks all for the feedback.

> The situation we have on the PTR: the system is waiting for 6 players: 2 melee, 2 ranged, and 2 healers. The team is still easy to predict: the first melee + the first caster + the first healer versus the second ones.

I know, but I hope it's still an improvement compared to the current status in the main realm.

> fixing this issue by implementing a rule such as 'no talent points = can't queue' would be a good solution.

I have enabled a new feature in the PTR, so now the players can't join without talents or without items equipped.

> Low levels are displaying in arena queue but arena does not pop with them

We should show only the last 70+ or we could show only the players in queue of the players level (ex. if you are 10-19 you'll see only the players queued for 10-19).

**Comment by Reily93** [comment]
if we made a spread sheet with all the possible combinations of classes that can play toighter for a working and functional setup with spec would help?

**Reply by Helias** [reply]
not sure, it might help but we need also a fallback plan wether these combinations are not matched.
I am wondering if these feature that I have enabled in the PTR are stable enough to put them into the cc live and improve the current status of the 3v3soloQ

**Reply by lamsxx** [reply]
Could you put them on the live server, please? If there are any other things to report, we will let you know as we go. The changes made on the PTR regarding the soloQ balancing are a good start compared to what's currently on the live server. Thank you in advance. 

Biti.

**Reply by Helias** [reply]
Done, at the next restart these features will be on cc live, thanks for all the feedback

**Comment by minthes** [comment]
To insure a fair queue, I suggest adding some grief protection in the following way: If someone doesn't accept the invite to the arena or leaves before the doors open, the game automatically draws as soon as the gates open. 
In a draw, you don't gain or lose any rating/mmr and the game is not even counted.

**Reply by Helias** [reply]
it's not a bad idea but there is always a margin of abuse here, if someone already knows that the team will lose, he will quit
we should at least remove some points to who leaves, I don't remember right now if we remove the rating or not if someone leave before the start.

Anyways, it's a fair suggestion

**Reply by Apathyxia** [reply]
I agree with both here. Only the person that leaves the arena before the doors open or doesn't accept the queue should face a rating decrease, but everyone else should be +0.

**Reply by minthes** [reply]
> it's not a bad idea but there is always a margin of abuse here, if someone already knows that the team will lose, he will quit we should at least remove some points to who leaves, I don't remember right now if we remove the rating or not if someone leave before the start.
> 
> Anyways, it's a fair suggestion

Sorry I should clarify. I don't mean if someone leaves during a match. I meant before the game starts. If we're still in preparation room and someone decides to leave, the match should be a draw. If they leave during the game, It's a normal game and not a draw.

**Reply by armchairrockstar1** [reply]
> Sorry I should clarify. I don't mean if someone leaves during a match. I meant before the game starts. If we're still in preparation room and someone decides to leave, the match should be a draw. If they leave during the game, It's a normal game and not a draw.

It's common practice to check /who to see opponents before gates are opened so this can be abused.

**Reply by minthes** [reply]
![image](https://github.com/user-attachments/assets/333f94c8-cf8f-4030-bf0a-08e3730e656d)
seems like this is a problem facing players.
It's not fair to lose a game and rating because someone else missed their pop and you're forced to 2v3.
It should be if someone misses their queue the game ends in a draw before the gates open.

**Reply by Helias** [reply]
Thanks for the comment, from the next restart this will be applied, the "incomplete game" will be disabled, and the rating should not be affected.

**Comment by Apathyxia** [comment]
It seems like the update for queues to pop only when there's 2 healers, 2 melees, and 2 ranged in queue is now in place. Having 10 people in the queue and it not popping is a bit meh. Could there be introduced a timer after which it would force pop the queue?

**Reply by Ottawia** [reply]
Makeing it 2dps+heal vs 2dps+heal could help a bit with that. Especialy when 2 mele comps and 2 casters comps can be really good.
![WoWScrnShot_010225_232817](https://github.com/user-attachments/assets/1977a1fc-70d8-4be3-9aa7-a055acaffc73)

**Reply by Franaloudzx** [reply]
**Quote: "Makeing it 2dps+heal vs 2dps+heal could help a bit with that."**

1- I don’t dislike the idea but (there is a big **BUT**): i would discard it **until they manage to fix the second biggest problem that we were facing: "Class Stacking"**.  Implementing 2 dps + heal having this problem would again create a terrible mixture of comps.

2.- To develope more my preview point, the good thing that also comes from the new system brought yesterday, is that the "Class Stacking" is reduced drastically by it’s own nature: basically the only classes that can stack now are classes with Dps/Healer talents choice. The rest will avoid this issue being a positive side effect.

3- That being said, the balance would be worse: **double caster is more likely to win against double melee**, this is a fact.

4.- It’s not even been 24 hours … and you guys speak about queue not popping, compared to what scenario…? The terrible system that we had before…? Of course, but we are here to get a real SoloQ 3vs3 being playable, **don't forget please that players can also queue normal arenas and get their fast pops**, with the main difference that in normal arena bracket, you are choosing freely to play with whatever you want, now you are playing a legitimate SoloQ 3vs3 system instead of the non-sense that we had previously. Sharing my experience: yesterday (a “Thursday” with all that implies) i had more than enough games. That taking into account that this change **didn’t have a single announcement about it’s release so people probably will be unaware  of it until Tuesday** (day when most players log to do their arena games).

5.- At this point, i’m gonna give a little reminder, people can opt to play normal arena brackets 2vs2 / 3vs3, but honestly i didn't see any problem in getting almost persistenly queue pop, all taking into consideration what this change is about, creating a normal arena experience not the joke that we previously had (that btw could hardly be called SoloQ).

Sincerely

Franagan

**Comment by minthes** [comment]
The queue seems to be bugged. If you insta queue after your game ends, it'll pop for you, regardless of how many people should be in queue before you. The problem of class stacking also persists

**Reply by Apathyxia** [reply]
I can confirm that this is how it works currently. 
Last week, I faced the issue of being in the queue for 30 minutes plus and missing 10 queues when people coming out from the games got pops before me. This week another person observed the same. 

It only made it pop for me when I was THE ONLY ONE in the queue, and 5 more people queued.

**Reply by Franaloudzx** [reply]
Yeah, it happened the same to me, i was the 1st in queue as DPS talents and i didn't get any pop either. However, that didn't happen when i was with HEALER talents. It's strange.

**Comment by armchairrockstar1** [comment]
Today everyone waited a lot in the queue. All matches I did were 2dps+1healer vs 2dps+1healer. Even when there could be 3dps vs. 3dps. But I think the rules about caster/melee dps made it impossible. I suggest removing melee/caster distinction as it's neither feasible nor reasonable.

**Comment by Never6** [comment]
the server population is too small to have "balanaced" teams
old system was better, with this current system you have many situations where multiple players sit in queue and are unable to play

**Reply by Never6** [reply]
1 healer per team is fine, its not what i would prefer given the population but this 1 melee 1 caster limit per team makes it impossible for players to play sometimes unless they respec (and possible grief games due to playing an alt over their main)
not to mention there are many real comps that include 2x melee or 2x caster
mage lock 
ele boomie
ret rogue
ret war
dk war 
enh war
etc

**Comment by Apathyxia** [comment]
I don't know if it's gonna help anything, but last week I made a report on the queue logic behaviour: 
https://github.com/azerothcore/mod-arena-3v3-solo-queue/issues/29

This is still happening, and it's so easily abused to keep getting instant pops for you while several others get skipped. 

TL;DR: If you exit the arena match before everyone else (i.e., match closes), and instantly queue, you will retain your priority in the queue and will be put on the top of the queue again. 

Maybe someone sees this as I think not many go to look at the module.

**Comment by armchairrockstar1** [comment]
Que not popping. Remove caster vs melee restrictions and allow 3dps vs. 3dps matches please.
![Screenshot (1119)](https://github.com/user-attachments/assets/2a8674ad-50b8-4652-a9b9-5e85677c3ee5)

**Comment by Ottawia** [comment]
MMR might be bugged in solo queue and affect the amount of rating gained.
There are two pictures of my two characters:
![priest](https://github.com/user-attachments/assets/4b1b9d61-8fc7-42bc-b11a-4b32ffa23fc2)
![druid](https://github.com/user-attachments/assets/01d287ec-df75-429a-aa3c-a910037d105b)
So what exactly is the problem here? Why does my druid, with less than a 50% win ratio (46/110⋅100%=~42%), have a higher MMR than my priest, who has over a 60% win ratio (190/287⋅100%=~66%)? It can't be caused by previous arena seasons because I had a higher rating on my priest in all brackets in all seasons.
Another thing—why are my priest's rating and personal rating two different numbers? Until yesterday, when I checked those numbers, I was sure that solo queue and personal solo queue ratings were always the same for one character.

**Reply by Apathyxia** [reply]
I agree, it is bugged. I have ~700 MMR, and rating of ~600. For example, I lost nearly 50 rating (also MMR) yesterday when I lost to a team with ratings of  ~1800, ~1800 and ~1600. 

My team had ratings of ~500, ~600 and  ~1600. 
When we won a similar match, I gained 5 rating. 

Something is clearly off, and it makes it very unmotivating to queue when even at low ratings you lose more than you get from a win (even against higher rated teams). The rating is more or less cosmetic, so I don't care about what that gives essentially, but I just like to see number go up.

**Comment by tbcxd** [comment]
Please make sure to reset mmr when new season begins & maybe consider allowing using Soloq rating to purchase gear that have rating requirements

**Comment by Ottawia** [comment]
Soloq is getting confused when players do not play last talents in trees (?)
Both druids from that game are resto and they are in same team
![image](https://github.com/user-attachments/assets/3127793e-6f7d-4ca0-b440-30279a295df2)

**Reply by Apathyxia** [reply]
Yup, can confirm. We had two resto druids in that game, and it's also my best guess that it is due to Agothoraz not having picked the last talent in the restoration tree.

**Comment by VasOsk1** [comment]
Any chance soloq will be fixed for melee? Some people just always skips other's in queue. It's nrly impossible to get in. Doesn't really matter if they join right after match ended or quit match and join in. You can sit in queue for lets say 15 min, new person join when there is no match atm and that person skip you. Already checked that. 

Example:

Melee's in queue:

AAA
BBB
CCC
DDD

EEE joins (no match ongoin atm)

AAA
BBB
EEE
CCC
DDD

No matter you do always EEE will get before CCC&DDD, so its only possible if caster dps and healer queue in before EEE queue. Yesterday was in queue 45 min, without even 1 game. There was always people that joined (no matter when) and was queued before me and few other melee's. That persons that skip queue are always the same.

It makes soloq totally unplayable.
